tests/unit/acquisition/test_rule.py

```diff
@@ -191,6 +192,25 @@ def test_discrete_thompson_sampling_acquire_returns_correct_shape(
     npt.assert_array_equal(query_points.shape, tf.constant([num_query_points, 2]))
 
 
+@pytest.mark.parametrize("num_query_points", [-1, 0])
+def test_random_sampling_raises_for_invalid_init_params(num_query_points: int) -> None:
+    with pytest.raises(ValueError):
+        RandomSampling(num_query_points)
+
+
+@pytest.mark.parametrize("num_query_points", [1, 10, 50])
+def test_random_sampling_acquire_returns_correct_shape(num_query_points: int) -> None:
+    search_space = Box([-2.2, -1.0], [1.3, 3.3])
+    rule = RandomSampling(num_query_points)
+    dataset = Dataset(tf.zeros([1, 2], dtype=tf.float64), tf.zeros([1, 1], dtype=tf.float64))
+    model = QuadraticMeanAndRBFKernelWithSamplers(
+        dataset=dataset, noise_variance=tf.constant(1.0, dtype=tf.float64)
+    )
+    query_points = rule.acquire(search_space, model)
```

Comment:
I get typing errors here that I'm not sure how to deal with, @uri-granta?
```
tests/unit/acquisition/test_rule.py:204: error: Need type annotation for "rule"  [var-annotated]
tests/unit/acquisition/test_rule.py:209: error: Argument 2 to "acquire" of "RandomSampling" has incompatible type "QuadraticMeanAndRBFKernelWithSamplers"; expected "Mapping[str, Any]"  [arg-type]
```

Reply:
You're should be calling acquire_single not acquire as you're passing in just a model, not a mapping of models (it happens to work because RandomSampling ignores that argument anyway). Not sure why the first error is happening though.